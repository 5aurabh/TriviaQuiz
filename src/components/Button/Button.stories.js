import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links';
import Button from './index.js';

storiesOf('Button', module)
  .add('default', () => <Button />)
  .add('custom text', () => <Button text="custom text" />)
  .add('loading', () => <Button isLoading={true} />)
  .add('disabled', () => <Button disabled={true} />)
  .add('custom color', () => <Button backgroundColor="green" />)
  .add('custom text color', () => <Button color="green" />);
