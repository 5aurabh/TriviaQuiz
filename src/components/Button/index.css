.Button {
  display: inline-block;
  color: white;
  border: 2px solid transparent;
  border-radius: 5px;
  font-size: 0.98rem;
  height: 55px;
  min-width: 120px;
  border: none;
  box-shadow: 1px 1px 0px 2px rgba (0, 0, 0, 0.3);
  cursor: pointer;
  text-align: center;
}

.smallButton {
  height: 35px;
  min-width: 90px;
  font-size: 0.8rem;
}

.Button:focus {
  outline: 0;
}

.slightlyRight {
  margin-right: 2px;
}
