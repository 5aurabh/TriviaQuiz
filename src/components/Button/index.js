// @flow
import React from 'react';
import styles from './index.css';
import { ClipLoader } from 'react-spinners';
type Props = {
  text: ?React.Node,
  loadingText: ?React.Node,
  loaderColor: ?string,
  isLoading: ?boolean,
  onClick: () => *,
  disabled: ?boolean,
  color: ?string,
  backgroundColor: string,
};
const Button = (props: Props) => (
  <button
    style={{
      backgroundColor: props.disabled ? 'grey' : props.backgroundColor,
      color: props.color,
      cursor: props.disabled ? 'not-allowed' : 'pointer',
    }}
    className={styles.Button}
    disabled={props.disabled || props.isLoading}
  >
    {props.isLoading && (
      <ClipLoader
        color={props.loaderColor}
        loading={true}
        sizeUnit={'px'}
        size={20}
        className={styles.slightlyRight}
      />
    )}

    {!props.isLoading ? (
      props.text
    ) : (
      <span
        style={{
          marginLeft: '5px',
          top: '-5px',
          position: 'relative',
        }}
      >
        {props.loadingText}
      </span>
    )}
  </button>
);

Button.defaultProps = {
  text: 'Sample button',
  loadingText: 'loading',
  backgroundColor: 'tomato',
  loaderColor: '#fff',
  color: '#fff',
  isLoading: false,
  disabled: false,
};

export default Button;
