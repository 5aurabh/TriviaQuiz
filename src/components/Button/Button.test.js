import React from 'react';
import renderer from 'react-test-renderer';
import Button from './index.js';
import { mount } from 'enzyme';

it('renders default button state', () => {
  const component = renderer.create(<Button />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders disabled button correctly', () => {
  const wrapper = mount(<Button disabled={true} />);
  const button = wrapper.find('button');
  console.log(button.props()['disabled']);
  expect(button.props()['disabled']).toEqual(true);
});
