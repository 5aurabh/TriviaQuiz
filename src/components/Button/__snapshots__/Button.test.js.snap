// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default button state 1`] = `
<button
  className={undefined}
  disabled={false}
  onClick={undefined}
  style={
    Object {
      "backgroundColor": "tomato",
      "color": "#fff",
      "cursor": "pointer",
    }
  }
>
  Sample button
</button>
`;
